lib/generate_puppetfile/bin.rb

```diff
@@ -292,6 +301,74 @@ def generate_module_data_from_Puppetfile
       modules
     end
 
+    # Public: generate the ad hoc repository (non-forge module) data from an existing Puppetfile
+    #   Returns an array of hashes with keys name, location, type, id
+    def generate_repository_data_from_Puppetfile
+      repositories = []
+
+      # Open the Puppetfile
+      File.open(@options[:puppetfile], 'r') do |fh|
+        while (line = fh.gets) != nil
+          # Skip blank lines, comments, anything that looks like a forge module
+          next if line =~ /^forge /
+          next if line =~ /^s+$|^#/
```

Comment:
Should this be `next if line =~ /^\s*$|^#/` ?  (`\s` instead of `s` and `*` instead of `+` to catch blank lines)?